pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,591 @@
+			let id = NextCollectionIdOf::<T>::get()
+				.or(T::CollectionId::initial_value())
+				.ok_or(NftsErrorOf::<T>::UnknownCollection)?;
+			NftsOf::<T>::create(origin, T::Lookup::unlookup(admin.clone()), config)?;
+			Self::deposit_event(Event::Created { id, creator, admin });
+			Ok(())
+		}
+
+		/// Destroy a collection of fungible items.
+		///
+		/// # Parameters
+		/// - `collection` - The collection to destroy.
+		/// - `witness` - Information on the items minted in the collection. This must be
+		/// correct.
+		#[pallet::call_index(8)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::destroy(
+    		witness.item_metadatas,
+    		witness.item_configs,
+    		witness.attributes,
+            witness.item_holders,
+            witness.allowances,
+		))]
+		pub fn destroy(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			witness: DestroyWitness,
+		) -> DispatchResultWithPostInfo {
+			NftsOf::<T>::destroy(origin, collection, witness)
+		}
+
+		/// Set an attribute for a collection or item.
+		///
+		/// # Parameters
+		/// - `collection` - The collection whose item's metadata to set.
+		/// - `maybe_item` - The item whose metadata to set.
+		/// - `namespace` - Attribute's namespace.
+		/// - `key` - The key of the attribute.
+		/// - `value` - The value to which to set the attribute.
+		#[pallet::call_index(12)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::set_attribute())]
+		pub fn set_attribute(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: Option<ItemIdOf<T>>,
+			namespace: AttributeNamespaceOf<T>,
+			key: BoundedVec<u8, T::KeyLimit>,
+			value: BoundedVec<u8, T::ValueLimit>,
+		) -> DispatchResult {
+			NftsOf::<T>::set_attribute(origin, collection, item, namespace, key, value)
+		}
+
+		/// Clear an attribute for the collection or item.
+		///
+		/// # Parameters
+		/// - `collection` - The collection whose item's metadata to clear.
+		/// - `maybe_item` - The item whose metadata to clear.
+		/// - `namespace` - Attribute's namespace.
+		/// - `key` - The key of the attribute.
+		#[pallet::call_index(13)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::clear_attribute())]
+		pub fn clear_attribute(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: Option<ItemIdOf<T>>,
+			namespace: AttributeNamespaceOf<T>,
+			key: BoundedVec<u8, T::KeyLimit>,
+		) -> DispatchResult {
+			NftsOf::<T>::clear_attribute(origin, collection, item, namespace, key)
+		}
+
+		/// Set the metadata for an item.
+		///
+		/// # Parameters
+		/// - `collection` - The collection whose item's metadata to set.
+		/// - `item` - The item whose metadata to set.
+		/// - `data` - The general information of this item. Limited in length by `StringLimit`.
+		#[pallet::call_index(14)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::set_metadata())]
+		pub fn set_metadata(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+			data: BoundedVec<u8, T::StringLimit>,
```

Comment:
Could this parameter be changed to Vec<u8> and we handle the length check in the runtime, so contract api implementation does not need to handle it?